homeassistant/components/airzone/climate.py

```diff
@@ -222,6 +223,9 @@ async def async_set_temperature(self, **kwargs: Any) -> None:
             params[API_HEAT_SET_POINT] = kwargs[ATTR_TARGET_TEMP_LOW]
         await self._async_update_hvac_params(params)
 
+        if ATTR_HVAC_MODE in kwargs:
+            await self.async_set_hvac_mode(kwargs[ATTR_HVAC_MODE])
```

Comment:
Does all params support all modes or we need to set the mode first?

Reply:
> Does all params support all modes or we need to set the mode first?

AFAIK yes, all params support all modes and you should be able to set the temperature while in modes such as FAN or DRY.